dbt/task/deps.py

```diff
@@ -14,250 +17,325 @@
-
-    def incorporate(self, package, version_specifiers=None):
-        if version_specifiers is None:
-            version_specifiers = [UnboundedVersionSpecifier()]
-        elif not isinstance(version_specifiers, list):
-            # error
+    def _sanitize_version(cls, version):
+        return cls.version_to_list(version) or ['master']
+
+    @property
+    def version(self):
+        return self._version
+
+    @version.setter
+    def version(self, version):
+        self._version = self._sanitize_version(version)
+
+    def incorporate(self, other):
+        return GitPackage(self.git, self.version + other.version)
+
+    def _resolve_version(self):
+        requested = set(self.version)
+        if len(requested) != 1:
+            logger.error(
+                'Could not resolve to a single version for Git repo %s!',
+                self.git)
+            logger.error('  Requested versions: %s', requested)
             raise Exception('bad')
-        else:
-            for version_specifier in version_specifiers:
-                if not isinstance(version_specifier, VersionSpecifier):
-                    # error
-                    raise Exception('bad')
+        self.version = requested.pop()
+
+    def _checkout(self, project):
+        """Performs a shallow clone of the repository into the downloads
+        directory. This function can be called repeatedly. If the project has
+        already been checked out at this version, it will be a no-op. Returns
+        the path to the checked out directory."""
+        if len(self.version) != 1:
+            dbt.exceptions.raise_dependency_error(
+                'Cannot checkout repository until the version is pinned.')
+        dir_ = dbt.clients.git.clone_and_checkout(
+            self.git, DOWNLOADS_PATH, branch=self.version[0],
+            dirname=self._checkout_name)
+        return os.path.join(DOWNLOADS_PATH, dir_)
+
+    def _fetch_metadata(self, project):
+        path = self._checkout(project)
+        with open(os.path.join(path, 'dbt_project.yml')) as f:
+            return load_yaml_text(f.read())
+
+    def install(self, project):
+        dest_path = self.get_installation_path(project)
+        if os.path.exists(dest_path):
+            shutil.rmtree(dest_path)
+        shutil.move(self._checkout(project), dest_path)
+
+
+class LocalPackage(Package):
+    def __init__(self, local):
+        super(LocalPackage, self).__init__(local)
+        self.local = local
+
+    def incorporate(self, _):
+        return LocalPackage(self.local)
+
+    def _fetch_metadata(self, project):
+        with open(os.path.join(self.local, 'dbt_project.yml')) as f:
+            return load_yaml_text(f.read())
+
+    def install(self, project):
+        dest_path = self.get_installation_path(project)
+        if os.path.exists(dest_path):
+            shutil.rmtree(dest_path)
+        shutil.copytree(self.local, dest_path)
+
+
+def _parse_package(dict_):
+    if dict_.get('package'):
```

Comment:
Can we also check that only one of these is specified? Something like this:

```
packages:
  - git: https://github.com/fishtown-analytics/dbt-utils
    hub: fishtown-analytics/redshift
    version: 1.0.0
```

Will probably be a common failure mode for people unfamiliar with yaml

Reply:
| Packages should not contain more than one of ['package', 'git', 'local']; yours has 2 of them - ['package', 'git']

 look ok?

Reply:
perfect